File: frontend/src/services/giderService.ts
import apiClient from './apiClient';
import { ApiResponse, SayfaliResponse } from '@/types/api.types';
import { GiderFiltre, GiderOlusturRequest, GiderResponse } from '@/types/gider.types';

export const giderService = {
  async listele(filtre: GiderFiltre = {}): Promise<ApiResponse<SayfaliResponse<GiderResponse>>> {
    const response = await apiClient.get<ApiResponse<SayfaliResponse<GiderResponse>>>('/giderler', { params: filtre });
    return response.data;
  },

  async getById(id: number): Promise<ApiResponse<GiderResponse>> {
    const response = await apiClient.get<ApiResponse<GiderResponse>>(`/giderler/${id}`);
    return response.data;
  },

  async olustur(data: GiderOlusturRequest): Promise<ApiResponse<GiderResponse>> {
    const response = await apiClient.post<ApiResponse<GiderResponse>>('/giderler', data);
    return response.data;
  },

  async guncelle(id: number, data: Partial<GiderOlusturRequest>): Promise<ApiResponse<GiderResponse>> {
    const response = await apiClient.put<ApiResponse<GiderResponse>>(`/giderler/${id}`, data);
    return response.data;
  },

  async sil(id: number): Promise<void> {
    await apiClient.delete(`/giderler/${id}`);
  },
};

File: backend/src/main/java/com/scinar/giderlerim/service/GiderService.java
package com.scinar.giderlerim.service;

import com.scinar.giderlerim.dto.request.GiderGuncelleRequest;
import com.scinar.giderlerim.dto.request.GiderOlusturRequest;
import com.scinar.giderlerim.dto.response.ApiResponse;
import com.scinar.giderlerim.dto.response.GiderResponse;
import com.scinar.giderlerim.dto.response.SayfaliResponse;
import com.scinar.giderlerim.entity.enums.OdemeYontemi;

import java.time.LocalDate;

public interface GiderService {
    ApiResponse<SayfaliResponse<GiderResponse>> listele(Long kullaniciId, Long kategoriId, LocalDate baslangic, LocalDate bitis, OdemeYontemi odemeYontemi, int page, int size);
    ApiResponse<GiderResponse> getById(Long kullaniciId, Long giderId);
    ApiResponse<GiderResponse> olustur(Long kullaniciId, GiderOlusturRequest request);
    ApiResponse<GiderResponse> guncelle(Long kullaniciId, Long giderId, GiderGuncelleRequest request);
    ApiResponse<Void> sil(Long kullaniciId, Long giderId);
}
